src/Microsoft.OData.Core/Json/ODataJsonOutputContextBase.cs

```diff
@@ -120,6 +122,17 @@ internal abstract class ODataJsonOutputContextBase : ODataOutputContext
         }
 
         /// <summary>
+        /// The output stream to write the payload to.
+        /// </summary>
+        internal Stream OutputStream

```

Comment:
>OutputStream [](start = 24, length = 12)

Rather than add a new property for OutputStream which is a duplicate of either messageOutputStream or asynchronousOutputStream, consider a GetMessageStream method that returns asynchronousOutputStream if not null, otherwise messageOutputStream.